openedx/core/djangoapps/user_api/tests/test_views.py

```diff
@@ -1405,6 +1468,56 @@ def test_register_with_profile_info(self):
         self.assertEqual(account_settings["goals"], self.GOALS)
         self.assertEqual(account_settings["country"], self.COUNTRY)
 
+    @override_settings(REGISTRATION_EXTENSION_FORM='openedx.core.djangoapps.user_api.tests.test_helpers.TestCaseForm')
+    @mock.patch('openedx.core.djangoapps.user_api.tests.test_helpers.TestCaseForm.DUMMY_STORAGE')
+    def test_with_extended_form(self, storage_dict):
+        test_dict = {}
+
+        def getter(name):
+            """
+            Get the value from the test dictionary.
+            """
+            return test_dict[name]
+
+        def setter(name, val):
+            """
+            Set the value in the test dictionary.
+            """
+            test_dict[name] = val
+
+        storage_dict.__getitem__.side_effect = getter
+        storage_dict.__setitem__.side_effect = setter
+        # Create a new registration
+        response = self.client.post(self.url, {
+            "email": self.EMAIL,
+            "name": self.NAME,
+            "username": self.USERNAME,
+            "password": self.PASSWORD,
+            "honor_code": "true",
+            "favorite_movie": "Inception",
+            "favorite_editor": "cat",
+        })
+        self.assertHttpOK(response)
+        self.assertIn(settings.EDXMKTG_LOGGED_IN_COOKIE_NAME, self.client.cookies)
+        self.assertIn(settings.EDXMKTG_USER_INFO_COOKIE_NAME, self.client.cookies)
+
+        user = User.objects.get(username=self.USERNAME)
+        request = RequestFactory().get('/url')
+        request.user = user
+        account_settings = get_account_settings(request)
+
+        self.assertEqual(self.USERNAME, account_settings["username"])
+        self.assertEqual(self.EMAIL, account_settings["email"])
+        self.assertFalse(account_settings["is_active"])
+        self.assertEqual(self.NAME, account_settings["name"])
+
+        # Verify that we've been logged in
+        # by trying to access a page that requires authentication
+        response = self.client.get(reverse("dashboard"))
+        self.assertHttpOK(response)
+
+        self.assertEqual(storage_dict[user.id], {'favorite_movie': "Inception", "favorite_editor": "cat"})
```

Comment:
Any reason you're not checking `storage_dict` on line 1513, but instead waiting until after the dashboard is loaded?